OpenRA.Mods.Common/Widgets/Logic/Ingame/GameInfoLogic.cs

```diff
@@ -90,7 +95,11 @@ public GameInfoLogic(Widget widget, ModData modData, World world, IngameInfoPane
 				if (tabButton != null)
 				{
 					tabButton.Text = info.Label;
-					tabButton.OnClick = () => activePanel = type;
+					tabButton.OnClick = () =>
+					{
+						panels[activePanel].Leave();
+						activePanel = type;
```

Comment:
Tbh this feels a bit overcomplicated, why not just
```suggestion
						if (activePanel == IngameInfoPanel.Chat)
							LeaveChatPanel();

						activePanel = type;
```
?